optuna/structs.py

```diff
@@ -135,6 +135,15 @@ def __ne__(self, other):
 
         return not self.__eq__(other)
 
+    def __lt__(self, other):
+        # type: (Any) -> bool
+
+        if not isinstance(other, type(self)):
+            raise TypeError('\'<\' not supported between instances of {} and {}'.format(
```

Comment:
`__lt__()` should return `NotImplemented` constants if it's not supported.
https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
Thank you for your suggestion. I've never known the usage of `NotImplemented`.
Fixed: https://github.com/optuna/optuna/pull/719/commits/e7a357802e555079bd9836ed0d1f83404ba17ca8